src/syntax/read.rkt

```diff
@@ -45,14 +45,25 @@
        stx)]
     ; special nullary operators
     [#`(,(or 'and 'or)) (datum->syntax #f 'TRUE stx)]
-    [#`(+) (datum->syntax #f 0 stx)]
-    [#`(*) (datum->syntax #f 1 stx)]
+    [#`(+) 
+      (warn 'nullary-operator "+ is deprecated as a nullary operator" )
+      (datum->syntax #f 0 stx)]
+    [#`(*)
+      (warn 'nullary-operator "* is deprecated as a nullary operator" )
+      (datum->syntax #f 1 stx)]
     ; special unary operators
-    [#`(,(or 'and 'or '+ '*) #,a) (expand a)]
+    [#`(,(or 'and 'or) #,a) (expand a)]
+    ; deprecated unary operators
+    [#`(,(or '+ '* '/) #,a)
+      (warn 'unary-operator "~a is deprecated as a unary operator" a) 
+      (expand a)]
     [#`(/ #,a) (datum->syntax #f (list '/ 1 (expand a)) stx)]
```

Comment:
This is not correct for `/`! `(/ a)` should not expand into `a`.

Reply:
Oh I see the other rule for that case anyway. I'll just remove the division from that condition assuming that we want to still allow (/ a) -> (1/a)

Reply:
Or should that also cause a warning.

Reply:
I'll add the warning.